ddl/index.go

```diff
@@ -908,27 +908,48 @@ func (w *addIndexWorker) run(d *ddlCtx) {
 
 func makeupDecodeColMap(sessCtx sessionctx.Context, t table.Table, indexInfo *model.IndexInfo) (map[int64]decoder.Column, error) {
 	cols := t.Cols()
-	decodeColMap := make(map[int64]decoder.Column, len(indexInfo.Columns))
-	for _, v := range indexInfo.Columns {
-		col := cols[v.Offset]
-		tpExpr := decoder.Column{
-			Col: col,
-		}
+	indexedCols := make([]*table.Column, len(indexInfo.Columns))
+	for i, v := range indexInfo.Columns {
+		indexedCols[i] = cols[v.Offset]
+	}
+
+	indexedColsCpy := make([]*table.Column, len(indexedCols))
+	copy(indexedColsCpy, indexedCols)
```

Comment:
Could you add some comment here, explain why we need a copied indexCols?

Reply:
@winkyao Because `buildFullDecodeColMap()` will modify it. Fine, let's move this logic into `buildFullDecodeColMap()` for better readability.